Records list broken

![screenshot from 2018-09-25 15-42-31](https://user-images.githubusercontent.com/546692/46018225-a98fa600-c0d9-11e8-8593-69e4c5694d1b.png)

```
index.js:150 Error compiling schema, function code: var refVal1 = refVal[1]; var validate =  (function  (data, dataPath, parentData, parentDataProperty, rootData) { 'use strict';  var vErrors = null;  var errors = 0;            var errs_1 = errors;    var errs_2 = errors; if ((typeof data !== "number" || (data % 1) || data !== data)) {  var err =  { keyword: 'type' , dataPath: (dataPath || '') + "" , schemaPath: '#/definitions/nonNegativeInteger/type' , params: { type: 'integer' }  , message: 'should be integer'  } ;  if (vErrors === null) vErrors = [err]; else vErrors.push(err); errors++;  }  if (typeof data === "number") {    if (  data < 0 || data !== data) {  var err =  { keyword: 'minimum' , dataPath: (dataPath || '') + "" , schemaPath: '#/definitions/nonNegativeInteger/minimum' , params: { comparison: '>=', limit: 0, exclusive: false }  , message: 'should be >= 0' } ;  if (vErrors === null) vErrors = [err]; else vErrors.push(err); errors++;  }   }  var valid2 = errors === errs_2;   var valid1 = errors === errs_1;   validate.errors = vErrors;  return errors === 0;        }); return validate;
w @ index.js:150

```

```
uncaught at u TypeError: Cannot read property 'startsWith' of undefined
    at v (https://settings-writer.stage.mozaws.net/v1/admin/static/js/main.0271a7c0.js:1:13008)

````

--- a/src/utils.js
+++ b/src/utils.js
@@ -233,7 +233,11 @@ export function buildAttachmentUrl(
   record: RecordData,
   capabilities: Capabilities
 ): ?string {
-  if (record.attachment == null || capabilities.attachments == null) {
+  if (
+    record.attachment == null ||
+    capabilities.attachments == null ||
+    !record.attachment.hasOwnProperty("location")
+  ) {
     return;
   }
   const { base_url = "" } = capabilities.attachments;

--- a/test/utils_test.js
+++ b/test/utils_test.js
@@ -184,12 +184,26 @@ describe("humanDate", () => {
 });
 
 describe("buildAttachmentUrl", () => {
+  const capabilities = { attachments: {} };
   it("should return nothing if no attachment is part of the record", () => {
-    expect(buildAttachmentUrl({}, { attachments: {} })).to.be.a("undefined");
+    expect(buildAttachmentUrl({ id: "abc" }, capabilities)).to.be.a(
+      "undefined"
+    );
+  });
+
+  it("should return nothing if the record attachment is incomplete", () => {
+    expect(buildAttachmentUrl({ attachment: {} }, capabilities)).to.be.a(
+      "undefined"
+    );
   });
 
   it("should return nothing if the attachments capability isn't enabled", () => {
-    expect(buildAttachmentUrl({ attachment: {} }, {})).to.be.a("undefined");
+    expect(
+      buildAttachmentUrl(
+        { id: "abc", attachment: { location: "http://" } },
+        { history: {} }
+      )
+    ).to.be.a("undefined");
   });
 
   it("should build an attachment url from the capability config", () => {

